modules/config/config.tf

```diff
@@ -3,8 +3,35 @@ resource "aws_config_configuration_recorder" "this" {
   role_arn = aws_iam_role.this.arn
 
   recording_group {
-    all_supported                 = true
-    include_global_resource_types = var.include_global_resource_types
+    all_supported                 = local.all_supported
+    include_global_resource_types = local.all_supported && var.include_global_resource_types
+
+    dynamic "exclusion_by_resource_types" {
+      for_each = length(var.exclude_resource_types) > 0 ? [1] : []
+      content {
+        resource_types = var.exclude_resource_types
+      }
+    }
+
+    resource_types = local.include_resource_types
+
+    recording_strategy {
+      use_only = length(local.include_resource_types) > 0 ? "INCLUSION_BY_RESOURCE_TYPES" : (
+        length(var.exclude_resource_types) > 0 ? "EXCLUSION_BY_RESOURCE_TYPES" : "ALL_SUPPORTED_RESOURCE_TYPES"
+      )
+    }
+  }
+
+  lifecycle {
+    precondition {
+      condition     = length(var.include_resource_types) != 0
```

Comment:
does this condition run every time? Kind of confused because it seems like this object would exist if `var.exclude_resource_types` is set instead of `var.include_resource_types` but then this condition would fail

Reply:
This ended up not being needed, I set it as a validation on the variable itself. We always expect the list to have elements.